tests/unit/store/tenant.spec.js

```diff
@@ -0,0 +1,369 @@
+        await store.dispatch('setCurrentTenant', 'team2')
+        expect(store.getters.tenant.slug).toEqual('team2')
+      })
+
+      it('should throw an error if the requested tenant does not exist', async () => {
+        expect(store.getters.tenant.slug).toBe(null)
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        expect(store.dispatch('setCurrentTenant', 'team3')).rejects.toThrow(
+          "Error: Unable to set current tenant: tenant doesn't exist"
+        )
+      })
+
+      it('should set the tenant role to Tenant Admin', async () => {
+        expect(store.getters.tenant.slug).toBe(null)
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        await store.dispatch('setCurrentTenant', 'team2')
+        expect(store.getters.role).toEqual('TENANT_ADMIN')
+      })
+
+      it('should return the tenant', async () => {
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        expect(await store.dispatch('setCurrentTenant', 'team2')).toEqual({
+          id: '12345',
+          name: 'boo',
+          role: 'TENANT_ADMIN',
+          slug: 'team2'
+        })
+      })
+    })
+
+    describe('setCurrentTenant - Cloud specific', () => {
+      let store
+
+      beforeEach(() => {
+        tenant.actions['auth0/updateAuthorization'] = jest.fn
+        tenant.actions['license/getLicense'] = jest.fn()
+        store = new Vuex.Store({
+          state: initialTenantState(),
+          getters: {
+            ...tenant.getters,
+            'api/isCloud': () => true,
+            'user/memberships': () => [
+              { tenant: { id: '12345' }, role: 'USER' }
+            ]
+          },
+          mutations: tenant.mutations,
+          actions: tenant.actions
+        })
+      })
+
+      it('should set the tenant role according to user memberships', async () => {
+        expect(store.getters.tenant.slug).toBe(null)
+        const tenantsArray = [{ name: 'boo', id: '12345', slug: 'team2' }]
+        prefectTenants.mockReturnValueOnce(tenantsArray)
+        await store.dispatch('setCurrentTenant', 'team2')
+        expect(store.getters.role).toEqual('USER')
+      })
+    })
+
+    describe('updateTenantSettings', () => {
+      let store
+
+      beforeEach(() => {
+        tenant.actions.getTenants = jest.fn()
+        store = new Vuex.Store({
+          state: initialTenantState(),
+          getters: tenant.getters,
+          mutations: tenant.mutations,
+          actions: tenant.actions
+        })
+      })
+
+      it('should throw an error if no settings are given', async () => {
+        expect(store.dispatch('updateTenantSettings')).rejects.toThrow(
+          'Error: passed invalid or empty settings object'
+        )
+      })
+
```

Comment:
Seems like we could add another test here for the update tenant settings mutation:
```
it ('should call the UpdateTenantSettings mutation with the passed settings', async () => {
  expect(mockedUpdateTenantSettings).toHaveBeenCalledWith({ variables: { name: 'Jerry' } })
})
```